irohad/consensus/yac/outcome_messages.hpp

```diff
@@ -15,50 +15,54 @@ namespace iroha {
   namespace consensus {
     namespace yac {
 
-      /**
-       * CommitMsg means consensus on cluster achieved.
-       * All nodes deals on some solution
-       */
-      struct CommitMessage {
-        explicit CommitMessage(std::vector<VoteMessage> votes)
-            : votes(std::move(votes)) {}
+      template <typename>
+      class OutcomeMessage {
+        std::string type_;
```

Comment:
this will copy name string on each construction. better
```suggestion
        virtual const std::string &getTypeName() const = 0;
```
and in implementations
```c++
        const std::string &getTypeName() const override { static const std::string name("Impl"); return name; }
```